rclcpp/src/rclcpp/expand_topic_or_service_name.cpp

```diff
@@ -105,17 +105,18 @@ rclcpp::expand_topic_or_service_name(
       if (validation_result == RCL_TOPIC_NAME_VALID) {
         throw std::runtime_error("topic name unexpectedly valid");
       }
-      const char * validation_message = rcl_topic_name_validation_result_string(validation_result);
-      if (!validation_message) {
-        throw std::runtime_error("unable to get validation error message");
-      }
-      if (is_service) {
-        using rclcpp::exceptions::InvalidServiceNameError;
-        throw InvalidServiceNameError(name.c_str(), validation_message, invalid_index);
-      } else {
-        using rclcpp::exceptions::InvalidTopicNameError;
-        throw InvalidTopicNameError(name.c_str(), validation_message, invalid_index);
+      eles {
```

Comment:
typo ?
This also need to be on the same line as the above closing curly bracket to pass the linters

Reply:
Also this else is not necessary as if we enter the if we throw an error so the previous code can be kept

Reply:
yes, it's a typo (my bad), actually the else got changed to not on the same line after I ran **ament_uncrusitify --reformat**, let me check.

why I add the else is because it will return `NULL` if `validation_result` is `RCL_TOPIC_NAME_VALID`, it will be problematic for `InvalidTopicNameError` and `InvalidServiceNameError`

Reply:
> why I add the else is because it will return NULL if validation_result is RCL_TOPIC_NAME_VALID, it will be problematic for InvalidTopicNameError and InvalidServiceNameError

My understanding is that this is taken care of by the if just before 
```
        if (validation_result == RCL_TOPIC_NAME_VALID) {
          throw std::runtime_error("topic name unexpectedly valid");
        }
```

Reply:
@gaoethan This comment has not been addressed. 
I think that the following snippet should be kept. If this evaluates to true the code below is unreacheable as we throw. If it doesnt we're sure we're not in the case of  `validation_result == RCL_TOPIC_NAME_VALID`
```c++
        if (validation_result == RCL_TOPIC_NAME_VALID) {
          throw std::runtime_error("topic name unexpectedly valid");
        }
```

Reply:
Actually, I have kept it, :) generally speaking, it has lower possibility to hit that and I put it in the [else statement](https://github.com/gaoethan/rclcpp/blob/4edc0930ca7dacbf3887d08f7920bbc19ca0b88b/rclcpp/src/rclcpp/expand_topic_or_service_name.cpp#L115-117)